tests/test_filtsmooth/test_gaussfiltsmooth/test_kalmanposterior.py

```diff
@@ -1,156 +1,185 @@
-            (len(loc_inputs[-1]), self.method.dynamics_model.dimension),
-        ]
-
-        for size in [(), (5,), (2, 3, 4)]:
-            for loc, loc_shape in zip(loc_inputs, single_sample_shapes):
-                with self.subTest(size=size, loc=loc):
-                    sample = self.posterior.sample(locations=loc, size=size)
-                    if size == ():
-                        self.assertEqual(sample.shape, loc_shape)
-                    else:
-                        self.assertEqual(sample.shape, size + loc_shape)
-
-    def test_sampling_all_locations_multiple_samples(self):
-        five_samples = self.posterior.sample(size=5)
-
-        chi_squared = np.array(
-            [
-                chi_squared_statistic(
-                    sample, self.posterior[:].mean, self.posterior[:].cov
-                )
-                for sample in five_samples
-            ]
-        ).mean()
-        self.assertLess(chi_squared, 10.0)
-        self.assertLess(0.1, chi_squared)
-
-    def test_sampling_two_locations_multiple_samples(self):
-        locs = self.posterior.locations[[2, 3]]
-        five_samples = self.posterior.sample(locations=locs, size=5)
-
-        chi_squared = np.array(
-            [
-                chi_squared_statistic(
-                    sample,
-                    self.posterior[:].mean[[2, 3]],
-                    self.posterior[:].cov[[2, 3]],
-                )
-                for sample in five_samples
-            ]
-        ).mean()
-        self.assertLess(chi_squared, 10.0)
-        self.assertLess(0.1, chi_squared)
-
-    def test_sampling_many_locations_multiple_samples(self):
-        locs = np.arange(0.0, 0.5, 0.025)
-        five_samples = self.posterior.sample(locations=locs, size=5)
-
-        chi_squared = np.array(
-            [
-                chi_squared_statistic(
-                    sample, self.posterior(locs).mean, self.posterior(locs).cov
-                )
-                for sample in five_samples
-            ]
-        ).mean()
-        self.assertLess(chi_squared, 10.0)
-        self.assertLess(0.1, chi_squared)
+from tests.testing import chi_squared_statistic
+
+from .filtsmooth_testcases import car_tracking
+
+
+@pytest.fixture
+def problem():
+    """Car-tracking problem."""
+    problem = car_tracking()
+    dynmod, measmod, initrv, info = problem
+
+    times = np.arange(0, info["tmax"], info["dt"])
+    states, obs = pnfss.generate(
+        dynmod=dynmod, measmod=measmod, initrv=initrv, times=times
+    )
+    return dynmod, measmod, initrv, info, obs, times, states
+
+
+@pytest.fixture
+def kalman(problem):
+    """Standard Kalman filter."""
+    dynmod, measmod, initrv, *_ = problem
+    return pnfs.Kalman(dynmod, measmod, initrv)
+
+
+@pytest.fixture
+def posterior(kalman, problem):
+    """Kalman filtering posterior."""
+    *_, obs, times, states = problem
+    return kalman.filter(obs, times)
+
+
+# This will raise a warning for the extrapolation test...
```

Comment:
Is there some way to catch that? Just because ideally, we'd like pytest not to return any warning at all..

Reply:
In principle, yes, but I am not sure currently how to best tie this to this specific instance of `posterior`. In fact, I am leaning more towards removing the warning than fixing this test (beyond the final data point, calling the posterior is pure extrapolation, which I think is as expected). What do you think?

Reply:
I removed the warning. Please complain if you think this was not a good idea. If not, please resolve :)